third_party/tonic/dart_wrappable.cc

```diff
@@ -22,13 +22,16 @@ Dart_Handle DartWrappable::CreateDartWrapper(DartState* dart_state) {
   Dart_PersistentHandle type = dart_state->class_library().GetClass(info);
   TONIC_DCHECK(!LogIfError(type));
 
-  intptr_t native_fields[kNumberOfNativeFields];
-  native_fields[kPeerIndex] = reinterpret_cast<intptr_t>(this);
-  native_fields[kWrapperInfoIndex] = reinterpret_cast<intptr_t>(&info);
-  Dart_Handle wrapper =
-      Dart_AllocateWithNativeFields(type, kNumberOfNativeFields, native_fields);
+  Dart_Handle wrapper = Dart_New(type, Dart_Null(), 0, nullptr);
```

Comment:
We cannot make objects that were previously constructible only by the engine to be also constructible by the user. This use will create objects that the engine cannot understand. Instead of making these constructors public, you can keep the constructor names the same and pass `_` (via a string handle) to the second argument to Dart_New.

Reply:
Yeah, I was trying to avoid that if I could in order to keep things more generic. I'll do that.

Reply:
Done.